Spatial module doesn't build

Environment: ubuntu 13.04 64-bit, OpenJDK 7, latest exist from git

Steps to reproduce:
1) Activate spatial module in $build-dir/extensions/indexes/local.build.properties
2) ./build.sh download-additional-jars
3) ./build.sh all

The end of the compile looks like this

```
compile-src:
     [echo] Compiling sources 'index-spatial'
    [mkdir] Created dir: /home/gtn/exist/extensions/indexes/spatial/build/classes
    [javac] Compiling 10 source files to /home/gtn/exist/extensions/indexes/spatial/build/classes
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 1.6
    [javac] /home/gtn/exist/extensions/indexes/spatial/src/org/exist/indexing/spatial/GMLHSQLIndexWorker.java:61: error: GMLHSQLIndexWorker is not abstract and does not override abstract method getQueryRewriter(XQueryContext) in IndexWorker
    [javac] public class GMLHSQLIndexWorker extends AbstractGMLJDBCIndexWorker {
    [javac]        ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] 1 error
    [javac] 1 warning

BUILD FAILED
/home/gtn/exist/build/scripts/build-impl.xml:415: The following error occurred while executing this line:
/home/gtn/exist/build/scripts/extensions-build.xml:31: The following error occurred while executing this line:
/home/gtn/exist/build/scripts/extensions-common.xml:137: The following error occurred while executing this line:
/home/gtn/exist/build/scripts/extensions-common.xml:199: The following error occurred while executing this line:
/home/gtn/exist/build/scripts/extensions-common.xml:152: The following error occurred while executing this line:
/home/gtn/exist/build/scripts/extensions-common.xml:163: Compile failed; see the compiler error output for details.
```

Also, note that the fetch url in $build-dir/extensions/indexes/spatial/extension.xml pointing to geoapi-nogenerics-2.1.1.jar (line 19) gets a 404. The file is available at the osgeo.org location used elsewhere in the file.

--- a/extensions/indexes/spatial/extension.xml
+++ b/extensions/indexes/spatial/extension.xml
@@ -14,9 +14,11 @@
 
  <target name="download-3rd-party" description="Download thirdparty jar files">
   <taskdef name="fetch" classname="nl.ow.dilemma.ant.fetch.FetchTask" classpathref="classpath.core"/>
+
+  <mkdir dir="${lib.dir}"/>
   
   <fetch classpathref="classpath.core" dest="${lib.dir}" usecache="true"
-   url="http://lists.refractions.net/m2/org/opengis/geoapi-nogenerics/2.1.1/geoapi-nogenerics-2.1.1.jar"
+   url="http://download.osgeo.org/webdav/geotools/org/opengis/geoapi-nogenerics/2.1.1/geoapi-nogenerics-2.1.1.jar"
    classname="org.opengis.geometry.Geometry"/>
   <fetch classpathref="classpath.core" dest="${lib.dir}" usecache="true"
    url="http://download.osgeo.org/webdav/geotools/org/geotools/gt2-api/2.4.5/gt2-api-2.4.5.jar"
@@ -49,7 +51,7 @@
    classname="javax.vecmath.GVector"/>
   
   <fetch classpathref="classpath.core" dest="${lib.dir}" 
-   url="http://sourceforge.net/projects/hsqldb/files/hsqldb/hsqldb_1_8_1/hsqldb_1_8_1_3.zip/download" classname="org.hsqldb.Database">
+   url="http://freefr.dl.sourceforge.net/project/hsqldb/hsqldb/hsqldb_1_8_1/hsqldb_1_8_1_3.zip" classname="org.hsqldb.Database">
    
    <patternset>
     <include name="**/hsqldb*.jar"/>

--- a/extensions/indexes/spatial/src/org/exist/indexing/spatial/GMLHSQLIndexWorker.java
+++ b/extensions/indexes/spatial/src/org/exist/indexing/spatial/GMLHSQLIndexWorker.java
@@ -34,17 +34,15 @@ import java.util.TreeMap;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.exist.collections.Collection;
-import org.exist.dom.persistent.DocumentImpl;
-import org.exist.dom.persistent.ExtArrayNodeSet;
-import org.exist.dom.persistent.NodeProxy;
-import org.exist.dom.persistent.NodeSet;
-import org.exist.dom.persistent.StoredNode;
+import org.exist.dom.persistent.*;
 import org.exist.indexing.spatial.AbstractGMLJDBCIndex.SpatialOperator;
 import org.exist.numbering.DLN;
 import org.exist.numbering.NodeId;
 import org.exist.security.PermissionDeniedException;
 import org.exist.storage.DBBroker;
+import org.exist.storage.NodePath;
 import org.exist.xmldb.XmldbURI;
+import org.exist.xquery.QueryRewriter;
 import org.exist.xquery.XPathException;
 import org.exist.xquery.value.AtomicValue;
 import org.exist.xquery.value.BooleanValue;
@@ -393,7 +391,7 @@ public class GMLHSQLIndexWorker extends AbstractGMLJDBCIndexWorker {
             " FROM " + GMLHSQLIndex.TABLE_NAME + 
             " WHERE DOCUMENT_URI = ? AND NODE_ID_UNITS = ? AND NODE_ID = ?;"
         );
-        ps.setString(1, p.getDocument().getURI().toString());
+        ps.setString(1, p.getOwnerDocument().getURI().toString());
         ps.setInt(2, p.getNodeId().units());
         byte[] bytes = new byte[p.getNodeId().size()];
         p.getNodeId().serialize(bytes, 0);
@@ -498,7 +496,7 @@ public class GMLHSQLIndexWorker extends AbstractGMLJDBCIndexWorker {
             " FROM " + GMLHSQLIndex.TABLE_NAME + 
             " WHERE DOCUMENT_URI = ? AND NODE_ID_UNITS = ? AND NODE_ID = ?"
         );
-        ps.setString(1, p.getDocument().getURI().toString());
+        ps.setString(1, p.getOwnerDocument().getURI().toString());
         ps.setInt(2, p.getNodeId().units());
         byte[] bytes = new byte[p.getNodeId().size()];
         p.getNodeId().serialize(bytes, 0);
@@ -731,7 +729,7 @@ public class GMLHSQLIndexWorker extends AbstractGMLJDBCIndexWorker {
                     return false;
                 }
                 NodeId nodeId = new DLN(rs.getInt("NODE_ID_UNITS"), rs.getBytes("NODE_ID"), 0);
-                StoredNode node = broker.objectWith(new NodeProxy(doc, nodeId));
+                IStoredNode node = broker.objectWith(new NodeProxy(doc, nodeId));
                 if (node == null) {
                     LOG.info("Node " + nodeId + "doesn't exist");
                     return false;
@@ -768,5 +766,15 @@ public class GMLHSQLIndexWorker extends AbstractGMLJDBCIndexWorker {
             if (ps != null)
                 ps.close();
         }
+    }
+
+    @Override
+    public <T extends IStoredNode> IStoredNode getReindexRoot(IStoredNode<T> node, NodePath path, boolean insert, boolean includeSelf) {
+        return null;
+    }
+
+    @Override
+    public QueryRewriter getQueryRewriter(XQueryContext context) {
+        return null;
     }
 }

